qase-javascript-commons/src/qase.ts

```diff
@@ -167,20 +167,46 @@ export class QaseReporter extends AbstractReporter {
     }
   }
 
+  /**
+   * @returns {Promise<void>}
+   */
+  public override async startTestRun(): Promise<void> {
+    if (!this.disabled) {
+
+      try {
+        await this.upstreamReporter?.startTestRun();
+      } catch (error) {
+        this.logError('Unable to start test run in the upstream reporter: ', error);
+
+        if (this.fallbackReporter == undefined) {
+          this.disabled = true;
+          return;
+        }
+
+        try {
+          await this.fallbackReporter?.startTestRun();
+        } catch (error) {
+          this.logError('Unable to start test run in the fallback reporter: ', error);
+          this.disabled = true;
+        }
+      }
+    }
+  }
+
   /**
    * @param {TestResultType} result
    */
-  public override addTestResult(result: TestResultType) {
+  public override async addTestResult(result: TestResultType) {
     if (!this.disabled) {
       this.logTestItem(result);
 
       if (this.useFallback) {
-        this.addTestResultToFallback(result);
+        await this.addTestResultToFallback(result);
         return;
       }
 
       try {
-        this.upstreamReporter?.addTestResult(result);
+        await this.upstreamReporter?.addTestResult(result);
```

Comment:
The function `addTestResultToFallback` returns nothing and is declared as `Promise<void>`, and this function also returns nothing but doesn't have a return type declared. What's the difference?

Reply:
I have updated a method's signature to async. In JS you can use `Promise<void>` function without await.